lint-configs/symlink-cpp-lint-configs.sh

```diff
@@ -15,7 +15,12 @@ symlink_config () {
     config_file_path="$1"
 
     repo_dir="$(git rev-parse --show-toplevel)"
-    repo_relative_config_file_path="$(realpath --relative-to="$repo_dir" "$config_file_path")"
+    config_file_absolute_path="$(readlink -f "$config_file_path")"
+
+    # NOTE: This is a bit fragile since it depends on adding a trailing slash to $repo_dir. Ideally,
+    # we would use `realpath --relative-to` instead of variable substitution, but `--relative-to`
+    # isn't available on macOS.
+    repo_relative_config_file_path="${config_file_absolute_path#"${repo_dir}/"}"
```

Comment:
Isn't adding the trailing slash the expected behavior? We could enforce it to be safe with:
```
repo_dir="${repo_dir%/}/"
````
and then modify repo_relative_config_file_path, src_path and dst_path to remove the added slash.

Reply:
I didn't remove the `/` from `src_path` and `dst_path` to keep them easier to read as paths (and the extra slash is harmless).